src/components/structures/RoomView.js

```diff
@@ -1663,14 +1676,16 @@ export default createReactClass({
         const ErrorBoundary = sdk.getComponent("elements.ErrorBoundary");
 
         if (!this.state.room) {
-            const loading = this.state.roomLoading || this.state.peekLoading;
+            const loading = this.state.matrixClientIsReady || this.state.roomLoading || this.state.peekLoading;
             if (loading) {
+                // Assume preview loading if we don't have a ready client or a room ID (still resolving the alias)
+                const previewLoading = !this.state.matrixClientIsReady || !this.state.roomId || this.state.peekLoading;
```

Comment:
What happened to the `roomLoading` flag?

Reply:
roomLoading was never part of `previewLoading`: 
https://github.com/matrix-org/matrix-react-sdk/pull/4606/files#diff-36759beafdbb5cc10e5d7e4ff1d98d87L1673